File: Android_Zone_Test/src/com/example/mylib_test/activity/system/ResultActivity2.java
package com.example.mylib_test.activity.system;

import com.example.mylib_test.R;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;

public class ResultActivity2 extends Activity implements OnClickListener {
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_result2);
	}

	@Override
	public void onClick(View v) {
		switch (v.getId()) {
		case R.id.bt_toThird:
			startActivityForResult(new Intent(this, ResultActivity3.class),SystemMainActivity.RequestCode);
			break;
		case R.id.bt_returnOne:
			setResult(SystemMainActivity.ResponseCode);
			finish();
			break;
			
		default:
			break;
		}
	}
	
	@Override
	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
		super.onActivityResult(requestCode, resultCode, data);
		System.out.println("second onActivityResult____ requestCode:"+requestCode+"\t resultCode"+resultCode);
	}
	

}

File: Android_Zone_Test/src/com/example/mylib_test/activity/system/ResultActivity2.kt
package com.example.mylib_test.activity.system

import com.example.mylib_test.R

import android.content.Intent
import android.view.View
import com.zone.lib.base.controller.activity.BaseFeatureActivity

class ResultActivity2 : BaseFeatureActivity() {
    override fun setContentView() {
        setContentView(R.layout.activity_result2)
    }

    override fun initData() {
    }

    override fun setListener() {
    }


    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.bt_toThird -> startActivityForResult(Intent(this, ResultActivity3::class.java), SystemMainActivity.RequestCode)
            R.id.bt_returnOne -> {
                setResult(SystemMainActivity.ResponseCode)
                finish()
            }

            else -> {
            }
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        println("second onActivityResult____ requestCode:$requestCode\t resultCode$resultCode")
    }
}
